http-client-javanet/src/main/java/io/micronaut/http/client/javanet/JavanetBlockingHttpClient.java

```diff
@@ -0,0 +1,91 @@
+import io.micronaut.http.client.HttpClientConfiguration;
+import io.micronaut.http.client.HttpVersionSelection;
+import io.micronaut.http.client.LoadBalancer;
+import io.micronaut.http.client.exceptions.HttpClientException;
+import io.micronaut.http.client.exceptions.HttpClientExceptionUtils;
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import io.micronaut.http.codec.MediaTypeCodecRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.http.HttpResponse;
+
+/**
+ * {@link io.micronaut.http.client.HttpClient} implementation for {@literal java.net.http.*} HTTP Client.
+ * @author Sergio del Amo
+ * @since 4.0.0
+ */
+@Internal
+@Experimental
+public class JavanetBlockingHttpClient extends AbstractJavanetHttpClient implements BlockingHttpClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JavanetBlockingHttpClient.class);
+
+    public JavanetBlockingHttpClient(
+        LoadBalancer loadBalancer,
+        HttpVersionSelection httpVersion,
+        HttpClientConfiguration configuration,
+        String contextPath,
+        MediaTypeCodecRegistry mediaTypeCodecRegistry,
+        RequestBinderRegistry requestBinderRegistry,
+        String clientId,
+        ConversionService conversionService,
+        JavanetClientSslBuilder sslBuilder
+    ) {
+        super(LOG, loadBalancer, httpVersion, configuration, contextPath, mediaTypeCodecRegistry, requestBinderRegistry, clientId, conversionService, sslBuilder);
+    }
+
+    @Override
+    public <I, O, E> io.micronaut.http.HttpResponse<O> exchange(io.micronaut.http.HttpRequest<I> request,
+                                              Argument<O> bodyType,
+                                              Argument<E> errorType) {
+        var httpRequest = mapToHttpRequest(request, bodyType).blockFirst();
+        try {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Client {} Sending HTTP Request: {}", clientId, httpRequest);
+            }
+            HttpResponse<byte[]> httpResponse = client.send(httpRequest, HttpResponse.BodyHandlers.ofByteArray());
+            boolean errorStatus = httpResponse.statusCode() >= 400;
+            if (errorStatus && configuration.isExceptionOnErrorStatus()) {
+                if (LOG.isErrorEnabled()) {
+                    LOG.error("Client {} Received HTTP Response: {} {}", clientId, httpResponse.statusCode(), httpResponse.uri());
+                }
+                throw HttpClientExceptionUtils.populateServiceId(new HttpClientResponseException(HttpStatus.valueOf(httpResponse.statusCode()).getReason(), response(httpResponse, bodyType)), clientId, configuration);
+            }
+            return response(httpResponse, bodyType);
+        } catch (IOException | InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new HttpClientException("error sending request", e);
+        }
```

Comment:
you shouldn't interrupt the thread on `IOException` only. Also include the message from the original exception in the message

Reply:
Fixed (I believe) by https://github.com/micronaut-projects/micronaut-core/pull/8441/commits/a53d3a81b06a96d8ee2dc2a76d7d0f1f8f117a79